src/db/database.go

```diff
@@ -33,40 +33,126 @@ func (d *DatabasePrisma) Disconnect() {
-		Image.Tags.Link(
-			Tag.TagName.In(tags),
-		),
+	return d.client.Tag.CreateOne(
+		Tag.Text.Set(tagName),
 	).Exec(ctx)
+}
+
+//GetAllTags .
+func (d *DatabasePrisma) GetAllTags() ([]TagModel, error) {
+	ctx := context.Background()
+	tags, err := d.client.Tag.FindMany().Exec(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	return nil
+	return tags, nil
 }
 
-//CreateTag .
-func (d *DatabasePrisma) CreateTag(tagName string) error {
+//FetchTags searchs for tags using theirs unique name
+func (d *DatabasePrisma) FetchTags(names []string) ([]TagModel, error) {
 	ctx := context.Background()
-	_, err := d.client.Tag.CreateOne(
-		Tag.TagName.Set(tagName),
+	return d.client.Tag.FindMany(
+		Tag.Text.In(names),
 	).Exec(ctx)
+}
+
+//FetchOrCreateTags searchs for tags (and creates them, if nedeed) using theirs unique name
+func (d *DatabasePrisma) FetchOrCreateTags(names []string) (map[string]TagModel, error) {
+	tags, err := d.FetchTags(names)
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	//Inserting into a map to reduce sort and search overhead
+	nTags := make(map[string]TagModel, len(tags))
+	for _, tag := range tags {
+		nTags[tag.Text] = tag
 	}
-	return nil
+
+	for _, name := range names {
+		_, ok := nTags[name]
+		if !ok {
+			newTag, err := d.CreateTag(name)
+			if err != nil {
+				return nil, err
+			}
+			nTags[name] = newTag
+		}
+	}
+
+	return nTags, nil
 }
 
-//GetAllTags .
-func (d *DatabasePrisma) GetAllTags() ([]TagModel, error) {
+//DeleteTag remove an existing database tag
+func (d *DatabasePrisma) DeleteTag(name string) error {
+	_, err := d.client.Tag.FindOne(
+		Tag.Text.Equals(name),
+	).Delete().Exec(context.Background())
+
+	return err
+}
+
+//DropAllTags removes all existing database tags
+func (d *DatabasePrisma) DropAllTags() (int, error) {
 	ctx := context.Background()
-	tags, err := d.client.Tag.FindMany().Exec(ctx)
+	return d.client.Tag.FindMany().Delete().Exec(ctx)
+}
+
+//DropAllItems removes all existing database items
+func (d *DatabasePrisma) DropAllItems() (int, error) {
+	ctx := context.Background()
+	return d.client.Item.FindMany().Delete().Exec(ctx)
+}
+
+//CreateItem creates a new item using provided information and new tags if needed.
+//It tries to reuse existing tag with the same name
+func (d *DatabasePrisma) CreateItem(title, subtitle, content string, tags []string) (ItemModel, error) {
+
+	mTags, err := d.FetchOrCreateTags(tags)
 	if err != nil {
-		return nil, err
+		return ItemModel{}, err
 	}
-	return tags, nil
+
+	item, err := d.client.Item.CreateOne(
+		Item.Title.Set(title),
+		Item.Subtitle.Set(subtitle),
+		Item.ContentURL.Set(content),
+		Item.Active.Set(true),
+	).Exec(context.Background())
+
+	if err != nil {
+		return item, err
+	}
+
+	//Link tags - TODO: Check if its possible to use/create one method to link them all
+	findResult := d.client.Item.FindOne(
+		Item.ID.Equals(item.ID),
+	)
+	for _, mTag := range mTags {
+		item, err = findResult.Update(
```

Comment:
I didn't test it, but I think *err* variable would have the last error occurred at findResult.Update(), so if occurred 3 errors, we would only return the last one. By the way, I don't see much problems in this situation. :slightly_smiling_face:

Reply:
Good point! I don't foresee anything that could break it, but I think we can raise a new issue about it, what do you think?

Reply:
Yes, good idea, we can create a new issue for this situation. =]